projects/packages/forms/tests/php/contact-form/Feedback_Test.php

```diff
@@ -1773,4 +1773,43 @@ public function test_has_field_type_legacy_feedback() {
 		$this->assertTrue( $response->has_field_type( 'consent' ), 'Legacy feedback should report consent field exists' );
 		$this->assertFalse( $response->has_consent(), 'Legacy consent should default to false (no)' );
 	}
+
+	public function test_get_field_by_id_and_value_by_id_new_submission() {
+		$form_id    = Utility::get_form_id();
+		$_post_data = Utility::get_post_request(
+			array(
+				'name'    => 'John Doe',
+				'email'   => 'john@example.com',
+				'message' => 'Hello!',
+			),
+			'g' . $form_id
+		);
+
+		$form = new Contact_Form(
+			array(
+				'title'       => 'Test Form',
+				'description' => 'This is a test form.',
+			),
+			"[contact-field label='Name' type='name' required='1'/][contact-field label='Email' type='email' required='1'/][contact-field label='Message' type='textarea' required='1'/]"
```

Comment:
It would be good to check - [contact-field label='Name' type='name' required='1' id="foo" /] 
And 

that 
```
$this->assertEquals( 'John Doe', $response->get_field_value_by_id( 'foo' ) );
```

Reply:
This is apparently not that simple. As written above, this will not pass. Our testing code (if not our production code) seems to be set up in a way where id=foo will be converted to something like gXXX-ID-foo. And then there's some kind of disconnect in the data. I'm having a hard time figuring out how to fix/resolve it in a non-artificial way.

Reply:
@enejb push a fix for this. Thanks!